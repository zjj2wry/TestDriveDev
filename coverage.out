TestDriveDev/tableTriveTest/word/word.go:8.34,10.22 2 1
TestDriveDev/tableTriveTest/word/word.go:15.2,15.25 1 1
TestDriveDev/tableTriveTest/word/word.go:20.2,20.13 1 1
TestDriveDev/tableTriveTest/word/word.go:10.22,11.26 1 1
TestDriveDev/tableTriveTest/word/word.go:11.26,13.4 1 1
TestDriveDev/tableTriveTest/word/word.go:15.25,16.46 1 1
TestDriveDev/tableTriveTest/word/word.go:16.46,18.4 1 1
