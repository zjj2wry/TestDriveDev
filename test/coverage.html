
<!DOCTYPE html>
<html>
	<head>
		<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
		<style>
			body {
				background: black;
				color: rgb(80, 80, 80);
			}
			body, pre, #legend span {
				font-family: Menlo, monospace;
				font-weight: bold;
			}
			#topbar {
				background: black;
				position: fixed;
				top: 0; left: 0; right: 0;
				height: 42px;
				border-bottom: 1px solid rgb(80, 80, 80);
			}
			#content {
				margin-top: 50px;
			}
			#nav, #legend {
				float: left;
				margin-left: 10px;
			}
			#legend {
				margin-top: 12px;
			}
			#nav {
				margin-top: 10px;
			}
			#legend span {
				margin: 0 5px;
			}
			.cov0 { color: rgb(192, 0, 0) }
.cov1 { color: rgb(128, 128, 128) }
.cov2 { color: rgb(116, 140, 131) }
.cov3 { color: rgb(104, 152, 134) }
.cov4 { color: rgb(92, 164, 137) }
.cov5 { color: rgb(80, 176, 140) }
.cov6 { color: rgb(68, 188, 143) }
.cov7 { color: rgb(56, 200, 146) }
.cov8 { color: rgb(44, 212, 149) }
.cov9 { color: rgb(32, 224, 152) }
.cov10 { color: rgb(20, 236, 155) }

		</style>
	</head>
	<body>
		<div id="topbar">
			<div id="nav">
				<select id="files">
				
				<option value="file0">TestDriveDev/mock/calculator/calculator.go (100.0%)</option>
				
				<option value="file1">TestDriveDev/mock/example/example.go (50.0%)</option>
				
				<option value="file2">TestDriveDev/mock/example/example_mock.go (100.0%)</option>
				
				<option value="file3">TestDriveDev/tableTriveTest/word/word.go (100.0%)</option>
				
				</select>
			</div>
			<div id="legend">
				<span>not tracked</span>
			
				<span class="cov0">no coverage</span>
				<span class="cov1">low coverage</span>
				<span class="cov2">*</span>
				<span class="cov3">*</span>
				<span class="cov4">*</span>
				<span class="cov5">*</span>
				<span class="cov6">*</span>
				<span class="cov7">*</span>
				<span class="cov8">*</span>
				<span class="cov9">*</span>
				<span class="cov10">high coverage</span>
			
			</div>
		</div>
		<div id="content">
		
		<pre class="file" id="file0" style="display: none">package calculator

type Random interface {
        Random(limit int) int
}

type Calculator interface {
        Add(x, y int) int
        Subtract(x, y int) int
        Multiply(x, y int) int
        Divide(x, y int) int
        Random() int
}

func newCalculator(rnd Random) Calculator <span class="cov10" title="5">{
        return calc{
                rnd: rnd,
        }
}</span>

type calc struct {
        rnd Random
}

func (c calc) Add(x, y int) int <span class="cov1" title="1">{
        return x + y
}</span>

func (c calc) Subtract(x, y int) int <span class="cov1" title="1">{
        return x - y
}</span>

func (c calc) Multiply(x, y int) int <span class="cov1" title="1">{
        return x * y
}</span>

func (c calc) Divide(x, y int) int <span class="cov1" title="1">{
        return x / y
}</span>

func (c calc) Random() int <span class="cov1" title="1">{
        return c.rnd.Random(100)
}</span>
</pre>
		
		<pre class="file" id="file1" style="display: none">package mock

import (
        "errors"
        "fmt"
)

type Object interface {
        DoSomething(number int) (bool, error)
        DoSomething2(name string) (ReturnObject, error)
}

type MyObject struct {
}

type ReturnObject struct {
        name string
}

func (m *MyObject) DoSomething(number int) (bool, error) <span class="cov0" title="0">{
        if number &gt; 0 </span><span class="cov0" title="0">{
                return true, nil
        }</span>
        <span class="cov0" title="0">return false, errors.New("game over")</span>
}

func (m *MyObject) DoSomething2(name string) (ReturnObject, error) <span class="cov0" title="0">{
        return ReturnObject{name}, nil
}</span>

func targetFuncThatDoesSomethingWithObj(m Object, number int) <span class="cov8" title="1">{
        result, err := m.DoSomething(number)
        fmt.Println(result, err)
}</span>

func targetFuncThatDoesSomethingWithObj2(m Object, name string) <span class="cov8" title="1">{
        result, err := m.DoSomething2(name)
        fmt.Println(result, err)
}</span>
</pre>
		
		<pre class="file" id="file2" style="display: none">package mock

import "github.com/stretchr/testify/mock"

var _ Object = new(Mock)

type Mock struct {
        mock.Mock
}

func (m *Mock) DoSomething(number int) (bool, error) <span class="cov8" title="1">{
        args := m.Called(number)
        return args.Bool(0), args.Error(1)

}</span>

func (m *Mock) DoSomething2(name string) (ReturnObject, error) <span class="cov8" title="1">{
        args := m.Called(name)
        return args.Get(0).(ReturnObject), args.Error(1)
}</span>
</pre>
		
		<pre class="file" id="file3" style="display: none">// Package word provides utilities for word games.
package word

import "unicode"

// IsPalindrome reports whether s reads the same forward and backward.
// Letter case is ignored, as are non-letters.
func IsPalindrome(s string) bool <span class="cov5" title="13">{
        var letters []rune
        for _, r := range s </span><span class="cov10" title="162">{
                if unicode.IsLetter(r) </span><span class="cov9" title="132">{
                        letters = append(letters, unicode.ToLower(r))
                }</span>
        }
        <span class="cov5" title="13">for i := range letters </span><span class="cov9" title="115">{
                if letters[i] != letters[len(letters)-1-i] </span><span class="cov2" title="3">{
                        return false
                }</span>
        }
        <span class="cov5" title="10">return true</span>
}
</pre>
		
		</div>
	</body>
	<script>
	(function() {
		var files = document.getElementById('files');
		var visible;
		files.addEventListener('change', onChange, false);
		function select(part) {
			if (visible)
				visible.style.display = 'none';
			visible = document.getElementById(part);
			if (!visible)
				return;
			files.value = part;
			visible.style.display = 'block';
			location.hash = part;
		}
		function onChange() {
			select(files.value);
			window.scrollTo(0, 0);
		}
		if (location.hash != "") {
			select(location.hash.substr(1));
		}
		if (!visible) {
			select("file0");
		}
	})();
	</script>
</html>
