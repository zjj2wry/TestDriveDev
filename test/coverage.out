TestDriveDev/tableTriveTest/word/word.go:8.34,10.22 2 13
TestDriveDev/tableTriveTest/word/word.go:15.2,15.25 1 13
TestDriveDev/tableTriveTest/word/word.go:20.2,20.13 1 10
TestDriveDev/tableTriveTest/word/word.go:10.22,11.26 1 162
TestDriveDev/tableTriveTest/word/word.go:11.26,13.4 1 132
TestDriveDev/tableTriveTest/word/word.go:15.25,16.46 1 115
TestDriveDev/tableTriveTest/word/word.go:16.46,18.4 1 3
